src/utils.ts

```diff
@@ -99,6 +99,23 @@ function display(value: unknown) {
   return typeof value === 'string' ? value : stringify(value);
 }
 
+export function stringifyDefined(input: unknown) {
+  if (input == null || typeof input !== 'object' || Array.isArray(input)) {
+    return stringify(input);
+  }
+
+  const output: Record<string, unknown> = {};
+  Object.keys(input).forEach((key) => {
+    // @ts-expect-error: input is an object
+    if (input[key] !== undefined) {
+      // @ts-expect-error: input is an object
+      output[key] = input[key];
+    }
```

Comment:
maybe we should add some js doc to explain what this function does and also that it only removes first level undefined keys wdyt?